packages/asana/test/adaptor.test.js

```diff
@@ -0,0 +1,206 @@
+import { expect } from 'chai';
+
+import {
+  execute,
+  request,
+  getTask,
+  getTasks,
+  updateTask,
+  createTask,
+  upsertTask,
+} from '../src';
+import { DEFAULT_PAGE_LIMIT } from '../src/util';
+import { mockServer } from './helper';
+
+const configuration = {
+  token: 'fake-token',
+};
+const state = { configuration };
+
+describe('Adaptor Test', () => {
+  describe('execute', () => {
+    it('executes each operation in sequence', done => {
+      let state = {};
+      let operations = [
+        state => {
+          return { counter: 1 };
+        },
+        state => {
+          return { counter: 2 };
+        },
+        state => {
+          return { counter: 3 };
+        },
+      ];
+
+      execute(...operations)(state)
+        .then(finalState => {
+          expect(finalState).to.eql({ counter: 3 });
+        })
+        .then(done)
+        .catch(done);
+    });
+
+    it('assigns references, data to the initialState', () => {
+      let state = {};
+
+      let finalState = execute()(state);
+
+      execute()(state).then(finalState => {
+        expect(finalState).to.eql({ references: [], data: null });
+      });
+    });
+  });
+
+  describe('request', () => {
+    it('throws if an absolute URL is passed', async () => {
+      // happily the request won't actually be made, so we don't need to mock anything here
+      let err;
+      try {
+        await execute(request('https://www.blah.com/a/b/c'))({});
+      } catch (e) {
+        err = e;
+      }
+
+      expect(err.code).to.equal('UNEXPECTED_ABSOLUTE_URL');
+    });
+  });
+
+  describe('getTask', () => {
+    it('should fetch a task by GID', async () => {
+      const taskGid = '12345';
+      const params = { opt_fields: 'name,notes' };
+      const mockData = { gid: taskGid, name: 'Test Task', notes: 'Some notes' };
+      mockServer
+        .intercept({
+          path: `/api/1.0/tasks/${taskGid}`,
+          query: params,
+          method: 'GET',
+        })
+        .reply(200, {
+          data: mockData,
+        });
+      const { data } = await getTask(taskGid, params)(state);
+
+      expect(data).to.eql(mockData);
+    });
+  });
+
+  describe('getTasks', () => {
+    it('should fetch tasks for a project', async () => {
+      const projectGid = 'proj123';
+      const params = { opt_fields: 'name' };
+      const mockData = [
+        { gid: '1', name: 'Task 1' },
+        { gid: '2', name: 'Task 2' },
+      ];
+      mockServer
+        .intercept({
+          path: `/api/1.0/projects/${projectGid}/tasks`,
+          query: { limit: DEFAULT_PAGE_LIMIT, ...params },
```

Comment:
Why is limit passed here, and why is it set to the default? It confuses the test